sdk/src/main/java/com/mapbox/maps/MapboxMap.kt

```diff
@@ -1704,6 +1688,9 @@ class MapboxMap :
    * @param mapProjection [MapProjection] to be applied to the map
    */
   @MapboxExperimental
+  @Deprecated(
+    "Use Projection style extension instead.",
```

Comment:
```suggestion
    "Use style extension instead: `loadStyle(styleUri) { +projection(projectionName) }`",
```

Reply:
Also `getMapProjection` should be deprecated.

Reply:
if that's the very first style load - then it will always be mercator.
if that's not the very first style load - `Style` object should be already available from previous style load